yellow-paper/docs/public-vm/avm-circuit.md

```diff
@@ -0,0 +1,40 @@
+---
+sidebar_position: 1
+---
+
+# AVM Circuit
+
+## Memory
+**VM memory** refers to the VM circuit's top-level memory structure. Since the VM circuit executes an entire public execution request, VM memory contains a dedicated region for each of a request's message calls.
+
+**Call memory** refers to a message call's dedicated memory region. Each call's memory region is of the same size `CALL_MEMORY_SIZE` and starts at offset `callMemoryOffset = callPointer * CALL_MEMORY_SIZE`.
```

Comment:
This is not how we model memory over different calls. Each call spawns a different instance of VM memory. They live in parallel universes so to speak.

The model is rather, "VM memory" is a layout which is instantiated for each call. An instance is annotated by a tag/identifier. In this way, we can address different call memory region in parallel.

Basically, we do not concatenate the address space, but a "global" address (callID, addr) where addr is u32.